src/home/getters/get-github-user.ts

```diff
@@ -30,13 +27,38 @@ async function getNewSessionToken(): Promise<string | null> {
   const params = new URLSearchParams(hash.substr(1)); // remove the '#' and parse
   const providerToken = params.get("provider_token");
   if (!providerToken) {
-    return null;
+    const err = params.get("error");
+    const code = params.get("error_code");
+    const desc = params.get("error_description");
+
+    if (err === "server_error") {
+      if (code === "500") {
+        if (desc === "Error getting user profile from external provider") {
+          // without a token we can't get a dynamic retry timeframe
+          displayPopupMessage(`GitHub Login Provider`, `Your access token may have reached it's rate limit, please try again after one hour.`);
+          throw new Error("GitHub login provider");
+        }
+      }
+    }
   }
   return providerToken;
```

Comment:
This defaults to null or undefined? I suspect undefined.

Reply:
> (method) URLSearchParams.get(name: string): string | null

but I've explicitly returned `token || null`